clippy_lints/src/utils/hir.rs

```diff
@@ -145,7 +145,7 @@ impl<'a, 'tcx: 'a> SpanlessEq<'a, 'tcx> {
             (&PatKind::TupleStruct(ref lp, ref la, ls), &PatKind::TupleStruct(ref rp, ref ra, rs)) => {
                 self.eq_path(lp, rp) && over(la, ra, |l, r| self.eq_pat(l, r)) && ls == rs
             }
-            (&PatKind::Ident(ref lb, ref li, ref lp), &PatKind::Ident(ref rb, ref ri, ref rp)) => {
```

Comment:
You might also want to add `PatKind::Path` here and add some tests for it. Something like:

```
if let None = Some(42) {
} else if let None = Some(42) {
}
```

See https://github.com/rust-lang/rust/pull/31685.